### PR TITLE
feat(cli): add unique-cli cite command for file page citation declarations [UN-21284]

### PR DESCRIPTION
**Pull request by benhaj**
## Summary

- Adds a lightweight `unique-cli cite` command that declares page-level file citations without extracting text. The agent reads files using its native capabilities (vision, pdftotext, etc.) and calls `cite` to register `[filesourceN]` markers.
- Writes entries to `.unique/file-refs.jsonl` manifest (same format consumed by the runner in monorepo).
- Includes deduplication within a turn, `--pages` range parsing, and `--local` flag for local files.

## Jira

[UN-21284](https://unique-ch.atlassian.net/browse/UN-21284)

## Companion PR

Monorepo PR (runner stitching pipeline): created separately.

## Test plan

- [x] Unit tests for `_parse_pages` (ranges, comma-separated, invalid inputs, edge cases)
- [x] Unit tests for `cmd_cite_file` (local files, KB resolution, dedup, numbering continuity, error handling)
- [ ] Manual E2E: agent reads a file, calls `cite`, response renders reference chips

Made with [Cursor](https://cursor.com)

[UN-21284]: https://unique-ch.atlassian.net/browse/UN-21284?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: unique_sdk/unique_sdk/cli/commands/cite_file.py [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+    state: ShellState, name_or_id: str
+) -> tuple[str, str]:
+    """Resolve a filename/content_id checking the chat-files manifest first.
+
+    Resolution order:
+    1. If name_or_id starts with "cont_", return it directly.
+    2. Check .unique/chat-files.json for a matching filename (exact or basename).
+    3. Fall back to KB resolution via _resolve_content_id.
+    """
+    if name_or_id.startswith("cont_"):
```
Comment (benhaj) [inline unique_sdk/unique_sdk/cli/commands/cite_file.py:70]: - Check if  name_or_id can really get a “cont_” entry 
Otherwise this is not needed

Comment (benhaj) [reply unique_sdk/unique_sdk/cli/commands/cite_file.py:70]: From the `ls` command output — When the agent runs unique-cli ls, it sees:
FILE  report.pdf  cont_abc123  1.2MB  2026-05-15

So the agent can actually decide to use the content id directly to cite and not use the filename.

File: unique_sdk/unique_sdk/cli/commands/cite_file.py [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+from unique_sdk.cli.state import ShellState
+
+CITE_ERROR_PREFIX = "cite:"
+
+_FILE_REFS_LOG_RELATIVE_PATH = Path(".unique") / "file-refs.jsonl"
+_FILE_REFS_LOCK_FILENAME = "file-refs.lock"
+_CHAT_FILES_MANIFEST = Path(".unique") / "chat-files.json"
+
+
+_MAX_PAGES_PER_CALL = 500
```
Comment (mfadler) [inline unique_sdk/unique_sdk/cli/commands/cite_file.py:24]: What happens for docs above 500 pages?

Comment (benhaj) [reply unique_sdk/unique_sdk/cli/commands/cite_file.py:24]: The 500 cap is per-call, not per-file. The agent can do cite file.pdf --pages 1-500 then cite file.pdf --pages 501-700. The guidance doesn't explicitly mention this because practically, citing 500+ individual pages in a single response is unrealistic — the agent typically cites 3-10 specific pages.

**Review by mfadler — APPROVED** [review]
